packages/dashboard/src/components/tasks/task-inspector.tsx

```diff
@@ -0,0 +1,243 @@
+
+export interface TableDataGridState {
+  task: TaskState | null;
+  open: boolean;
+  onClose: () => void;
+}
+
+export function TaskInspector({ task, open, onClose }: TableDataGridState): JSX.Element {
+  const theme = useTheme();
+  const rmf = React.useContext(RmfAppContext);
+  const appController = React.useContext(AppControllerContext);
+
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [taskLogs, setTaskLogs] = React.useState<TaskEventLog | null>(null);
+  const [waypoint, setWaypoint] = React.useState({
+    start: '',
+    finish: '',
+  });
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = AppEvents.taskSelect.subscribe((task) => {
+      if (!task) {
+        setTaskState(null);
+        setTaskLogs(null);
+        return;
+      }
+      (async () => {
+        try {
+          const logs = (
+            await rmf.tasksApi.getTaskLogTasksTaskIdLogGet(
+              task.booking.id,
+              `0,${Number.MAX_SAFE_INTEGER}`,
+            )
+          ).data;
+          setTaskLogs(logs);
+
+          if (task.phases) {
+            /**
+             * We assumed that phases[index].detail always return 
+             * "Go to " + goal_name_,
+                "Moving the robot from " + start_name + " to " + goal_name_,
+                *estimate); 
+
+                According with this code
+             * https://github.com/open-rmf/rmf_task/blob/main/rmf_task_sequence/src/rmf_task_sequence/events/GoToPlace.cpp#L252
+             */
+            const firstPhase = Object.values(task.phases)[0]?.detail;
+            const lastPhase = Object.values(task.phases).pop()?.detail;
+
+            if (!firstPhase) {
+              setWaypoint({
+                ...waypoint,
+                start: '',
+                finish: '',
+              });
+              return;
+            }
+
+            const colonIndex = firstPhase.toString().indexOf(':');
+            const closingBracketIndex = firstPhase.toString().indexOf(']');
+
+            setWaypoint({
+              ...waypoint,
+              start: firstPhase.toString().substring(colonIndex + 1, closingBracketIndex),
+              finish: lastPhase ? lastPhase.toString().replace(/^.+:/, '').replace(/.$/, '') : '',
+            });
+          }
+        } catch {
+          console.log(`Failed to fetch task logs for ${task.booking.id}`);
+          setTaskLogs(null);
+        }
+        setTaskState(task);
+      })();
+    });
+    return () => sub.unsubscribe();
+  }, [rmf, waypoint]);
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = AppEvents.taskSelect
+      .pipe(
+        switchMap((selectedTask) =>
+          selectedTask ? rmf.getTaskStateObs(selectedTask.booking.id) : of(null),
+        ),
+      )
+      .subscribe(setTaskState);
+    return () => sub.unsubscribe();
+  }, [rmf]);
+
+  React.useEffect(() => {
+    AppEvents.taskSelect.next(task);
+  }, [task]);
```

Comment:
Instead of setting up this workaround chain of `AppEvents.taskSelect`, I think we can make use of https://github.com/open-rmf/rmf-web/blob/main/packages/dashboard/src/components/rmf-app/rmf-ingress.ts#L201, subscribing directly to the ID of the `task` prop, and retrieving the logs using the same API call you used

Reply:
@aaronchongth the solution I did now make sense? Please correct me if my solution is not in line with your suggestion.

Reply:
your changes were exactly what I was considering, thanks for implementing it so quickly! Resolved! :partying_face: